stl/inc/type_traits

```diff
@@ -1845,7 +1845,11 @@ template <class _Ty, class _Uty>
 struct _Refwrap_has_ctor_from<_Ty, _Uty, void_t<decltype(_Refwrap_ctor_fun<_Ty>(_STD declval<_Uty>()))>> : true_type {};
 
 template <class _Ty>
-class reference_wrapper : public _Weak_types<_Ty>::type { // stand-in for an assignable reference
+class reference_wrapper
+#if !_HAS_CXX20
+    : public _Weak_types<_Ty>::type

```

Comment:
There's a design question here, because we haven't yet merged #380 to implement #28 (WG21-P0619 Removing C++17-Deprecated Features In C++20). Specifically, an escape hatch to restore `reference_wrapper`'s weak result type is incompatible with supporting incomplete types. There are two possible decisions: allow `reference_wrapper` to be affected by such an escape hatch (thereby disabling support for incomplete types), or exempt `reference_wrapper` from the escape hatch (so C++20 mode always activates support for incomplete types).

It's not a big deal because `reference_wrapper`'s support for weak result types is extremely obscure. I am inclined to accept the code as-is, meaning that C++20 mode always makes `reference_wrapper` support incomplete types and removes its weak result type, even if weak result types have otherwise been escape-hatched.